### PR TITLE
Feat/2465 pull service flow

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: #2465 

This PR's purpose is to build a service that consumes the other services that were built on the previous pr.
Built the flow and the logic of the service from the moment it gets the request from the client.

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.service.ts [hunk 1]
```diff
@@ -1,7 +1,190 @@
...
+        accessToken,
+      }
+    );
+
+    await this.gitClientService.clone(pushEventMessage, baseDir, accessToken);
+
+    this.logger.log(LoggerMessages.log.CLONE_SUCCESS, GitPullEventService.name);
+  }
+
+  private async managePullEventStorage(
```
Comment (mshidlov) [inline packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.service.ts:157]: managePullEventStorage is suggesting that this function does something with the storage - what it does with the storage is unclear....

this function brakes the single point of responsibility as it does copy and pull.

please rename the function for pull changes from existing or something that suggest the actual function role. you may think of a different solution as well.

File: packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.service.ts [hunk 2]
```diff
@@ -1,7 +1,190 @@
...
+      GitPullEventService.name
+    );
+  }
+
+  private async manageDelete(pullEventRecordId: bigint, dirToDelete: string) {
+    await this.gitPullEventRepository.update(
+      pullEventRecordId,
+      EnumGitPullEventStatus.Deleted
+    );
+    this.storageService.deleteDir(dirToDelete);
```
Comment (mshidlov) [inline packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.service.ts:183]:
```suggestion
    this.storageService.deleteDir(dirToDelete);
    await this.gitPullEventRepository.update(
      pullEventRecordId,
      EnumGitPullEventStatus.Deleted
    );
```

**Review by mshidlov — CHANGES_REQUESTED** [review]

File: packages/amplication-git-pull-service/.env [hunk 3]
```diff
@@ -0,0 +1,18 @@
+POSTGRESQL_USER=admin
+POSTGRESQL_PASSWORD=admin
+POSTGRESQL_PORT=5432
+POSTGRESQL_URL=postgres://admin:admin@localhost:5432
+BCRYPT_SALT=10
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/.env:5]: that's how it is on .env of the generated app

File: packages/amplication-git-pull-service/.env [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+COMPOSE_PROJECT_NAME=amplication-git-pull-service
+JWT_SECRET_KEY=Change_ME!!!
+JWT_EXPIRATION=2d
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/.env:8]: that's how it is on .env of the generated app

[the remark this answers is not in the document]
Comment (mshidlov) [reply packages/amplication-git-pull-service/.env:8]: OK We dont need JWT here it is part of the auth startegy

File: packages/amplication-git-pull-service/.env [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+NODE_ENV=production
+BROKER_IP_ENV_KEY=localhost:9092
```
[the remark this answers is not in the document]
Comment (mshidlov) [reply packages/amplication-git-pull-service/.env:11]: @abrl91 Kafka actually not suppose to be here at all

[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/.env:11]: the value? or at all?

[the remark this answers is not in the document]
Comment (mshidlov) [reply packages/amplication-git-pull-service/.env:11]: the environments are suppose to be part of the kafka PR and not the flow PR

[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/.env:11]: 😭

File: packages/amplication-git-pull-service/.env [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+CONSUMER_GROUP_ENV_ID=git-pull-event
+CLIENT_ID_ENV=repository-pull
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/.env:13]: so those either

File: packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.module.ts [hunk 7]
```diff
@@ -3,11 +3,42 @@ import { GitPullEventService } from "./gitPullEvent.service";
...
+import {
+  utilities as nestWinstonModuleUtilities,
+  WinstonModule,
+} from "nest-winston";
+import * as winston from "winston";
 
 @Module({
-  imports: [PrismaModule],
+  imports: [
+    WinstonModule.forRoot({
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/src/core/gitPullEvent/gitPullEvent.module.ts:18]: you said that I can do it on the next pr

File: packages/amplication-git-pull-service/src/providers/storage/storage.service.ts [hunk 8]
```diff
@@ -1,15 +1,20 @@
...
     try {
       await fse.copy(srcDir, destDir);
-    } catch (err) {
+    } catch (err: any) {
       throw new CustomError("failed to copy files from srcDir to destDir", err);
     }
   }
+
+  deleteDir(dir: string): void {
+    fs.rmSync(dir, { recursive: true, force: true });
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/src/providers/storage/storage.service.ts:18]: I didn't find something else that works with recursive

[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-git-pull-service/src/providers/storage/storage.service.ts:18]: `fs.rm(path, { recursive: true })`
is that OK?

**Pull request merged**